parser/ParseBlockProperties.hpp

```diff
@@ -0,0 +1,331 @@
+   */
+  bool hasNumericValue() const {
+    return property_ == Property::kBlockSize;
+  }
+
+ protected:
+  void getFieldStringItems(
+      std::vector<std::string> *inline_field_names,
+      std::vector<std::string> *inline_field_values,
+      std::vector<std::string> *non_container_child_field_names,
+      std::vector<const ParseTreeNode*> *non_container_child_fields,
+      std::vector<std::string> *container_child_field_names,
+      std::vector<std::vector<const ParseTreeNode*>> *container_child_fields) const override {
+    inline_field_names->push_back("property");
+    inline_field_values->push_back(getPropertyString());
+
+    if (hasStringValues()) {
+      container_child_field_names->push_back((values_->size() == 1) ? "value" : "values");
+      container_child_fields->emplace_back();
+      for (const ParseString& item_value : *values_) {
+        container_child_fields->back().push_back(&item_value);
+      }
+    } else {
+      container_child_field_names->push_back("value");
+      container_child_fields->emplace_back();
+      container_child_fields->back().push_back(numeric_value_.get());
+    }
+  }
+
+ private:
+  ParseBlockPropertyItem(int line_number,
+                         int column_number,
+                         Property property)
+      : ParseTreeNode(line_number, column_number),
+        property_(property),
+        values_(nullptr),
+        compress_all_(false),
+        numeric_value_(nullptr) {
+    if (hasStringValues()) {
+      values_.reset(new PtrList<ParseString>());
+    }
+  }
+
+  Property property_;
+  std::unique_ptr<PtrList<ParseString>> values_;
+  bool compress_all_;
+  std::unique_ptr<NumericParseLiteralValue> numeric_value_;
+
+  DISALLOW_COPY_AND_ASSIGN(ParseBlockPropertyItem);
+};
+
+/**
+ * @brief Contains user-specified physical properties of a table's blocks as BlockPropertyItems.
+ */
+class ParseBlockProperties : public ParseTreeNode {
+ public:
+  /**
+   * @brief Constructor.
+   *
+   * @param line_number Line number of the first token of this node in the SQL statement.
+   * @param column_number Column number of the first token of this node in the SQL statement.
+   **/
+  ParseBlockProperties(int line_number,
+                       int column_number,
+                       PtrList<ParseBlockPropertyItem>* properties)
+      : ParseTreeNode(line_number, column_number),
+        properties_(properties) { }
+
+  /**
+   * @return Name of the parser construct represented by this class.
+   */
+  std::string getName() const override {
+    return "BlockProperties";
+  }
+
+  /**
+   * @brief Gets the specified BlockPropertyItem.
+   *
+   * @param property The property, one of kSort, kCompress, or kType.
+   * @return Pointer to the property or nullptr if not found.
+   */
+  const ParseBlockPropertyItem* getPropertyItem(ParseBlockPropertyItem::Property property) const {
+    for (const ParseBlockPropertyItem &item : *properties_) {
+      if (item.property() == property) {
```

Comment:
Is there ever any validation to make sure that the same property is not specified twice? What about `COMPRESS`, where it is legal to specify more than one column that should be compressed (it seems like we only allow a single `COMPRESS` clause that optionally has comma-separated columns, which is OK, but we should probably have some validation to make sure that multiple clauses are superseding each other).